### PR TITLE
feat: Benchmarks + Github Action

### PR DESCRIPTION
**Pull request by james-milligan**

### CODE REVIEW

File: .github/workflows/publish-benchmark.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Trigger Site Rebuild on a CRON Schedule
+
+on:
+  schedule:
+    # Runs "at minute 55 past every hour" (see https://crontab.guru)
+    - cron: '0 2 * * *'
```
Comment (beeme1mr) [inline .github/workflows/publish-benchmark.yaml:6]: Is GMT used?

File: .github/workflows/publish-benchmark.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  benchmark:
+    name: Run Benchmark
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (beeme1mr) [inline .github/workflows/publish-benchmark.yaml:12]: Do you need to specify a branch or will it just checkout the default branch when using a schedule?

File: .github/workflows/publish-benchmark.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: Go Benchmark
+          tool: 'go'
+          output-file-path: output.txt
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
+          # Show alert with commit comment on detecting possible performance regression
+          alert-threshold: '120%'
+          comment-on-alert: true
```
Comment (beeme1mr) [inline .github/workflows/publish-benchmark.yaml:30]: Will this work since it's not a PR?

File: .github/workflows/publish-benchmark.yaml (hunk 1 above)
Comment (james-milligan) [reply .github/workflows/publish-benchmark.yaml:6]: Yes, I've updated the comment to reflect this + the correct time

File: .github/workflows/publish-benchmark.yaml (hunk 3 above)
Comment (james-milligan) [reply .github/workflows/publish-benchmark.yaml]: removed :)

**Comment by AlexsJones** [comment]
What does this PR do?

**Comment by james-milligan** [comment]
> What does this PR do?

https://github.com/open-feature/flagd/issues/134

Benchmark tests will be run daily and reported via GitHub pages as seen [here](https://github.com/benchmark-action/github-action-benchmark)
The stored performance values will then be compared against PRs so changes that affect performance can be highlighted 

I am keeping this as a draft until it is ready, as once [this PR](https://github.com/open-feature/flagd/pull/135) has resolved ill either need to cleanup the commits or refactor the tests

**Comment by james-milligan** [comment]
closing and reopening on repaired branch

**Pull request closed without merging**